connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -170,6 +170,7 @@ private <T> HttpResponse<T> httpRequest(HttpClient client, String url, String me
             }
         } catch (IOException | InterruptedException | TimeoutException | ExecutionException e) {
             log.error("IO error forwarding REST request: ", e);
+            Thread.currentThread().interrupt();
```

Comment:
Shouldn't we only do this if we've caught an `InterruptedException`? There are other potential exception types that could cause us to enter this code block right now.